spring-web/src/main/java/org/springframework/web/service/invoker/PathVariableArgumentResolver.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.web.service.invoker;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.core.MethodParameter;
+import org.springframework.core.ReactiveAdapter;
+import org.springframework.core.ReactiveAdapterRegistry;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.lang.Nullable;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+
+/**
+ * An implementation of {@link HttpServiceMethodArgumentResolver} that resolves request path variables
+ * based on method arguments annotated with {@link  PathVariable}. {@code null} values are allowed only if {@link PathVariable#required()}
+ * is {@code true}.
+ *
+ * @author Olga Maciaszek-Sharma
+ * @since 6.0
+ */
+public class PathVariableArgumentResolver implements HttpServiceMethodArgumentResolver {
```

Comment:
I see a number of highlights in IntelliJ for nullability that should be resolved.

Reply:
Will take a look. At some points, it was complaining about unnecessary null checks, but the tests have proven them necessary. Will verify anyway.